### PR TITLE
[#44] [#45] [UI] [Integrate] As a user, I can see the survey list is loading with skeleton animation in the Home screen

### LINKED ISSUE #44: [UI] As a user, I can see the survey list is loading with skeleton animation in the Home screen
## Why

When first entering the Home screen, we will need to wait for the Surveys to load, so to improve the UI/UX we should show the blank screen with skeleton animation in that time

## Acceptance Criteria

- Adapt Skeleton Animation as the design

## Design

CleanShot 2023-04-07 at 14 55 12
Link: https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?node-id=34-49&scaling=scale-down&page-id=0%3A1&starting-point-node-id=34%3A450&show-proto-sidebar=1

### LINKED ISSUE #45: [Integrate] As a user, I can see the survey list is loading with skeleton animation in the Home screen
## Why

When first entering the Home screen, we will need to wait for the Surveys to load, so to improve the UI/UX we should show the blank screen with skeleton animation in that time

## Acceptance Criteria

- When entering the Home screen for the first time, display the Skeleton Animation until all surveys are fully loaded

## Design

CleanShot 2023-04-07 at 14 55 12

Link: https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?node-id=34-49&scaling=scale-down&page-id=0%3A1&starting-point-node-id=34%3A450&show-proto-sidebar=1

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/44
https://github.com/markgravity/kmm-ic/issues/45

## What happened 👀

- Add `HomeSkeleton` to display a skeleton loading
- Integrate to show the `HomeSkeleton` when calling `HomeViewModel.fetch()`

## Insight 📝

n/a

## Proof Of Work 📹

https://github.com/markgravity/kmm-ic/assets/17875522/4820324c-2b29-4d64-8ea6-e5aa76d21cac

### CODE REVIEW

File: ios/KMMIC/Sources/Presentation/Modules/Home/Views/HomeSkeleton.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        Text(nil)
+            .skeleton(
+                with: true,
+                size: CGSize(
+                    width: width,
+                    height: (20.0 * CGFloat(lines)) + (6.0 * (CGFloat(lines) - 1))
+                )
+            )
+            .shape(type: .capsule)
+            .multiline(lines: lines, scales: [1: 0.5], spacing: 6.0)
```
Comment (phongvhd93) [inline ios/KMMIC/Sources/Presentation/Modules/Home/Views/HomeSkeleton.swift:64]: `scales: [1.0 : 0.5]

File: ios/KMMIC/Sources/Presentation/Modules/Home/Views/HomeSkeleton.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @ViewBuilder
+    func skeletonTextView(width: CGFloat, lines: Int = 1) -> some View {
+        Text(nil)
+            .skeleton(
+                with: true,
+                size: CGSize(
+                    width: width,
+                    height: (20.0 * CGFloat(lines)) + (6.0 * (CGFloat(lines) - 1))
```
Comment (phongvhd93) [inline ios/KMMIC/Sources/Presentation/Modules/Home/Views/HomeSkeleton.swift]: `height: (20.0 * CGFloat(lines)) + (6.0 * (CGFloat(lines) - 1.0))`

**Review by phongvhd93 — COMMENTED** [review]
This PR has some unrelated commits. Please check your commit history and rebase again

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/api/ApiService.kt
Comment (phongvhd93) [inline shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/api/ApiService.kt:1]: This file is unrelated, please revert

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/auth/AuthService.kt
Comment (phongvhd93) [inline shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/auth/AuthService.kt:1]: This file is unrelated, please revert

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/auth/model/AuthRefreshTokenParams.kt
Comment (phongvhd93) [inline shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/auth/model/AuthRefreshTokenParams.kt:1]: This file is unrelated, please revert

File: ios/KMMIC/Sources/Presentation/Modules/Home/Views/HomeSkeleton.swift (hunk 1 above)
Comment (markgravity) [reply ios/KMMIC/Sources/Presentation/Modules/Home/Views/HomeSkeleton.swift:64]: @phongvhd93 its type is `scales: [Int: CGFloat]`

File: ios/KMMIC/Sources/Presentation/Modules/Home/Views/HomeSkeleton.swift (hunk 2 above)
Comment (markgravity) [reply ios/KMMIC/Sources/Presentation/Modules/Home/Views/HomeSkeleton.swift]: @phongvhd93 updated at [e3e1b73](https://github.com/markgravity/kmm-ic/pull/108/commits/e3e1b73529aa33421b2d8101b239bdceb1ce9f7a)

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/api/ApiService.kt
Comment (markgravity) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/api/ApiService.kt:1]: Rebased 🙏

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/auth/AuthService.kt
Comment (markgravity) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/auth/AuthService.kt:1]: Rebased 🙏

File: shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/auth/model/AuthRefreshTokenParams.kt
Comment (markgravity) [reply shared/src/commonMain/kotlin/co/nimblehq/mark/kmmic/data/service/auth/model/AuthRefreshTokenParams.kt:1]: Rebased 🙏

**Review by phongvhd93 — APPROVED** [review]

**Review by suho — APPROVED** [review]
